libs/designsystem/src/lib/components/checkbox/checkbox.component.html

```diff
@@ -1,22 +1,10 @@
-<!-- Since there is a bug in Ionic when binding to the `mode` property of ion-checkbox,
-a switch between templates is used -->
-<ng-container *ngIf="shape === 'circle'; else squareBox">
+<div [id]="_labelId">
   <ion-checkbox
-    class="checkbox"
-    [ngClass]="classes"
+    mode="md"
     [checked]="checked"
     [disabled]="disabled"
     (ionChange)="onChecked($event.detail.checked)"
+    [attr.aria-labelledby]="labelText ? _labelId : ''"
   ></ion-checkbox>
-</ng-container>
-
-<ng-template #squareBox>
-  <ion-checkbox
-    class="checkbox"
-    [ngClass]="classes"
-    mode="md"
-    [checked]="checked"
-    (ionChange)="onChecked($event.detail.checked)"
-  >
-  </ion-checkbox>
-</ng-template>
+  {{ labelText }}
+</div>
```

Comment:
Should be conditionally wrapped in a kirby-label if `text` input property is set...

Reply:
I think this solution is simpler. Adding a wrapping kirby-label will only add more complication without any gains.
Semantically a kirby-label and ion-label do not add anything, as they do not contain an html label, and the Ionic getAriaLabel can't read a wrapping ion-label without a parent ion-item.